src/test/java/com/google/devtools/build/lib/blackbox/tests/manageddirs/ManagedDirectoriesBlackBoxTest.java

```diff
@@ -392,6 +392,50 @@ public void testWorkspaceSymlinkThrowsWithManagedDirectories() throws Exception
         .isTrue();
   }
 
+  @Test
+  public void testNoCheckExternalFiles() throws Exception {
+    testNoCheckFiles(false, false);
+  }
+  @Test
+  public void testNoCheckExternalFilesWatchFs() throws Exception {
+    testNoCheckFiles(false, true);
+  }
+  @Test
+  public void testNoCheckAllSkips() throws Exception {
+    testNoCheckFiles(true, false);
+  }
+  @Test
+  public void testNoCheckAllSkipsWatchFs() throws Exception {
+    testNoCheckFiles(true, true);
+  }
+
+  private void testNoCheckFiles(boolean allSkips, boolean watchFs) throws Exception {
+    generateProject();
+    buildExpectRepositoryRuleCalled(false, watchFs);
+    checkProjectFiles();
+
+    Path nodeModules = context().getWorkDir().resolve("node_modules");
+    assertThat(nodeModules.toFile().isDirectory()).isTrue();
+    PathUtils.deleteTree(nodeModules);
+    assertThat(nodeModules.toFile().isDirectory()).isFalse();
+
+    // We don't check the external file disappeared so the build is broken
+    BuilderRunner bazel = bazel(watchFs).withFlags("--noexperimental_check_external_files");
+    if (allSkips) {
+      bazel = bazel.withFlags("--noexperimental_check_output_files");
+    }
+    ProcessResult result = bazel.shouldFail().build("//...");
+    assertThat(findPattern(result, "Not found package.json")).isTrue();
+
+    // and it doesn't make the file
+    assertThat(nodeModules.toFile().isDirectory()).isFalse();
+
+    // but a normal build fixes it
+    // TODO: These don't check the directories and they should
```

Comment:
@janakdr here is the subsequent build that should re-scan the `node_modules` and fix it, but instead it just leaves it broken